symphony-bdk-http/symphony-bdk-http-jersey2/src/main/java/com/symphony/bdk/http/jersey2/ApiClientJersey2.java

```diff
@@ -187,18 +167,37 @@ public ApiClientJersey2(final Client httpClient, String basePath, Map<String, St
             buildResponseHeaders(response),
             respBody);
       }
+    }
+  }
+
+  private Response getResponse(Invocation.Builder invocationBuilder, String method, Entity<?> entity) throws ApiException {
+    try {
+      switch(method) {
+        case HttpMethod.GET:
+          return invocationBuilder.get();
+        case HttpMethod.POST:
+          return invocationBuilder.post(entity);
+        case HttpMethod.PUT:
+          return invocationBuilder.put(entity);
+        case HttpMethod.DELETE:
+          return invocationBuilder.method(HttpMethod.DELETE, entity);
+        case HttpMethod.PATCH:
+          return invocationBuilder.method(HttpMethod.PATCH, entity);
+        case HttpMethod.HEAD:
+          return invocationBuilder.head();
+        case HttpMethod.OPTIONS:
+          return invocationBuilder.options();
+        case "TRACE":
+          return invocationBuilder.trace();
+        default:
+          throw new ApiException(500, "unknown method type " + method);
+      }
     } catch (ProcessingException e) {
```

Comment:
@symphony-mariacristina can you check that it is was expected? Because to solve the conflict with your last commit I modified a lot...

Reply:
@symphony-enrico I made some tests and the ConnectTimeoutException is thrown by the invocationBuilder so we should be good with this refactoring. 👍